src/api/java/com/minecolonies/api/colony/buildings/IBuilding.java

```diff
@@ -77,10 +77,10 @@ public interface IBuilding extends IBuildingContainer, IRequestResolverProvider,
      * Get a module matching a certain predicate.
      * @param clazz the class of the module.
      * @param modulePredicate the predicate to match.
-     * @param <T> the optional type.
-     * @return optional of the matching predicate (could be empty).
+     * @param <T> the module type.
+     * @return the first matching module (could be null).
      */
-    @NotNull
+    @Nullable
```

Comment:
Then this needs to be one an optional.
It is either not null or returning an optional.....

From my side you can create an overload that returns the optional and throw an exception on the first if the module is not found.

Reply:
This is NotNull on purpose, we call this if we expect this to exist and throw an exception otherwise.
(AI dependent calls)

Reply:
Again, go look at the actual implementation.  The interface was lying.

Reply:
I know this, I implemented this. It's because it has to return something if it can't find it. But this should NEVER happen, because the place that calls it relies on it.

Reply:
That's the wrong way to do things.

Reply:
It's intentional, I don't want to do unpacking everywhere for something that cannot be null because it has to be there. For anything else there is the getOptional method.

Reply:
That's fine, but if you're going to do that then the annotation should still be correct.  Either it should be `@Nullable` because it can return null (and it's just the caller that's either expecting it or not expecting it, and the latter might give it an NPE), or it should be `@NotNullable` and it should throw an exception instead of returning null if it can't find what it's looking for.  There are no other correct options.

Reply:
And, FWIW, `IBuildingView.getModuleViewMatching()` is also @Nullable, and I didn't change that.  So this change makes it more consistent with that.

Reply:
Then make it throw an exception if not there